sagemaker_studio_image_build/codebuild.py

```diff
@@ -11,14 +11,17 @@
 
 
 class TempCodeBuildProject:
-    def __init__(self, s3_location, role, repository=None, compute_type=None):
+    def __init__(self, s3_location, role, repository=None, compute_type=None, environment=None):
         self.s3_location = s3_location
         self.role = role
 
         self.session = boto3.session.Session()
         self.domain_id, self.user_profile_name = self._get_studio_metadata()
         self.repo_name = None
         self.compute_type = compute_type or "BUILD_GENERAL1_SMALL"
+        self.environment = environment or "LINUX_CONTAINER"
+        if self.environment=="LINUX_GPU_CONTAINER" and self.compute_type=="BUILD_GENERAL1_SMALL":
+            self.compute_type="BUILD_GENERAL1_LARGE"
```

Comment:
Question: Why is this required?

Reply:
BUILD_GENERAL1_LARGE will automatically switch to a GPU instance if LINUX_GPU_CONTAINER is set. Small or medium will not.

Reply:
Thanks. In that case, I recommend we make this explicit - Fail if the user is requested GPU and Small or Medium, because these are incompatible configuration options